open.md

```diff
@@ -4,4 +4,112 @@ title: TOKI OSN Open Contest 2019 (English)
 key: open
 ---
 
-To Be Announced
+### Preface
+
+- **TOKI OSN Open Contest 2019** is an open competition mirroring the National Science Olympiad in Informatics 2019 in Indonesia (**OSN Informatika 2019** in Bahasa).
+- OSN Informatika is a national olympiad in informatics for high school students. It is the one of the stages required for Indonesian students to represent Indonesia in IOI (International Olympiad in Informatics)
+- The problemset is the same as the problemset in OSN Informatika 2019, and will be given in Bahasa Indonesia and English.
+- The contest will be held on [TLX (TOKI Online Judge)](https://tlx.toki.id/contests) (you may need to register for a TLX account if you do not have one already).
+- If you have any questions please contact [sc-osn-2019@googlegroups.com](mailto:sc-osn-2019@googlegroups.com).
+
+<br>
+
+### Rules Overview
+
+* There will be **one trial session** and **2 (two) competition days**.
+* Result of trial session does **not** affect scoring.
+* There will be **3 (three) problems** on each day to be solved in **5 (five) hours**.
+* Each contestant may submit up to **50 submissions** for each problem.
+* The supported programming languages are **C, C++, and Pascal**.
+* During the competition, each contestant can only see their own scores.
+
+<br>
+
+### Schedule and Technical Details
+
+- The contests will be held at the following time ranges:
+* **Day 0**: [2 July 2019 08:35 - 23:05 UTC](https://www.timeanddate.com/worldclock/fixedtime.html?msg=TOKI+OSN+Open+Contest+2019+Day+0&iso=20190701T1535&p1=108&ah=14&am=30) (trial session)
+* **Day 1**: [3 July 2019 02:35 - 23:05 UTC](https://www.timeanddate.com/worldclock/fixedtime.html?msg=TOKI+OSN+Open+Contest+2019+Day+1&iso=20190702T0935&p1=108&ah=20&am=30)
+* **Day 2**: [4 July 2019 02:35 - 23:05 UTC](https://www.timeanddate.com/worldclock/fixedtime.html?msg=TOKI+OSN+Open+Contest+2019+Day+2&iso=20190703T0935&p1=108&ah=20&am=30)
+
+- Each contestant can select **any 5-hour window** within the time range to do the contest for each competition day.
+- All three contests are now available on [TLX](https://tlx.toki.id/contests). Please register in those contests.
+- Each contestant may start the contest any time within the time range, by clicking the timer button.
+- There will be no additional time given if a contestant starts the contest in less than 5 hours before the contest ends.
+- The open contest doesn't support clarifications, but all important announcements from the original contest will be broadcast as well.
+
+<br>
+
+### Problems Types
+
+- There are 3 possible problem types. There may be some problem type that is not given.
```

Comment:
`s/There may be some problem type that is not given./Some problem type might not be tested.`